http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java

```diff
@@ -0,0 +1,119 @@
+package io.quarkus.ts.http.advanced.reactive;
+
+import static io.quarkus.ts.http.advanced.reactive.MediaTypeResource.APPLICATION_YAML;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import org.eclipse.microprofile.openapi.annotations.parameters.RequestBody;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+
+@Path("/city")
+public class CityResource {
+
+    private static final String YAML_FILE_PATH = "payload.yaml";
+
+    private final Logger logger = LoggerFactory.getLogger(CityResource.class);
+
+    private final ObjectMapper objectMapper = new YAMLMapper();
+
+    @Inject
+    private CityListWrapperSerializer serializer;
+
+    private List<City> cityList = new ArrayList<>();
+
+    @GET
+    @Produces(value = MediaType.APPLICATION_XML)
+    public CityListWrapper getCities() {
+        logger.info("Received request to getCities");
+
+        if (cityList.isEmpty()) {
+            cityList.add(new City("San Bernardino", "EEUU"));
+            cityList.add(new City("Brno", "Czech Republic"));
+            cityList.add(new City("Zaragoza", "Spain"));
+        }
+        return new CityListWrapper(cityList);
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_XML)
+    @Produces(value = MediaType.APPLICATION_XML)
+    public String createCity(String xml) {
+
+        logger.info("Received XML payload: {}", xml);
+
+        CityListDTO cityListDTO = serializer.fromXML(xml);
+
+        if (cityListDTO == null) {
+            logger.error("Error deserializing XML");
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("<error>Invalid XML payload</error>")
+                    .build().toString();
+        }
+
+        List<City> cities = cityListDTO.getCityList();
+
+        for (City city : cities) {
+            logger.info("Saving city {}", city);
+        }
+
+        String responseXML = serializer.toXML(cityListDTO);
+
+        return responseXML;
+    }
+
+    @POST
+    @Path("/cityYaml")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes("application/x-yaml")
+    public Object handleYamlPostRequest(@RequestBody final String yamlPayload) throws IOException {
```

Comment:
If this is `string`, there is no serialization in place, how does this actually test YAML content type consumption if you do serialization/deserialization yourself?

Please let me give you example: change `@Consumes` to JSON  or plain text and let me know if there is any difference. Thank you